src/libraries/System.Net.Security/src/System/Net/Security/Pal.Android/SafeFreeSslCredentials.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net
+{
+    internal sealed class SafeFreeSslCredentials : SafeFreeCredentials
+    {
+        public SafeFreeSslCredentials(SslStreamCertificateContext? certificateContext, SslProtocols protocols, EncryptionPolicy policy)
+            : base(IntPtr.Zero, true)
+        {
+            if (certificateContext != null)
+            {
+                // Make a defensive copy of the certificate. In some async cases the
```

Comment:
I'm not sure this is really needed but I know macOS has it there. So it is probably OK to keep.